macros/misc/randomNamesPronouns.pl

```diff
@@ -0,0 +1,662 @@
+
+    $p->subject;
+
+returns the pronoun. In this case 'he'.
+
+=cut
+
+sub subject {
+	return shift->{pronoun};
+}
+
+=head2 Subject
+
+This returns the subject pronoun as an upper case.
+
+    $p->Subject;
+
+returns the upper case pronoun. In this case 'He'.
+
+=cut
+
+sub Subject {
+	return ucfirst(shift->{pronoun});
+}
+
+=head2 possessive
+
+This returns the possessive adjective
+
+    $p->possessive;
+
+returns (his, her, their) for the pronouns (he/she/they)
+
+=cut
+
+sub possessive {
+	my $p = shift->{pronoun};
+	return $p eq 'he' ? 'his' : ($p eq 'she' ? 'her' : 'their');
+}
+
+=head2 Possessive
+
+This returns the captilized possessive adjective
+
+    $p->Possessive;
+
+returns (His, Her, Their) for the pronouns (he/she/they)
+
+=cut
+
+sub Possessive {
+	return ucfirst(shift->possessive);
+}
+
+=head2 possession
+
+This returns the possessive pronoun
+
+    $p->possession;
+
+returns (his, hers, theirs) for the pronouns (he/she/they)
+
+=cut
+
+sub possession {
+	my $p = shift->{pronoun};
+	return $p eq 'he' ? 'his' : ($p eq 'she' ? 'hers' : 'theirs');
+}
+
+=head2 Possession
+
+This returns the capitalized versions of possessive pronoun
+
+    $p->Possession;
+
+returns (His, Hers, Theirs) for the pronouns (he/she/they)
+
+=cut
+
+sub Possession {
+	return ucfirst(shift->possession);
+}
+
+=head2 object
+
+This returns the object pronoun
+
+    $p->object;
+
+returns (him, her, them) for the pronouns (he/she/they)
+
+=cut
+
+sub object {
+	my $p = shift->{pronoun};
+	return $p eq 'he' ? 'him' : ($p eq 'she' ? 'her' : 'them');
+}
+
+=head2 Object
+
+This returns the captilized object pronoun
+
+    $p->Object;
+
+returns (Him, Her, Them) for the pronouns (he/she/they)
+
+=cut
+
+sub Object {
+	return ucfirst(shift->object);
+}
+
+=head2 verb
+
+Returns the correct conjugation of the verb. If only one argument is passed in, it should
+be a regular verb in the third person plural conjugation (the "they" version). For example:
+
+    $p1 = new Person(name => 'Roger', pronoun => 'he');
+    $p1->verb('find');
+
+returns 'finds'. All that happens is an 's' or an 'es' is appended, according to some
+simple heuristics.
+
+    $p2 = new Person(name => 'Max', pronoun => 'they');
+    $p2->verb('find')
+
+returns 'find'
+
+If a second argument is passed, it should be the third person singular conjugation
+of the same verb (the "he/she" version). For example:
+
+    $p1 = new Person(name => 'Roger', pronoun => 'he');
+    $p1->verb('fly', 'flies');
+
+returns 'flies'
+
+    $p2 = new Person(name => 'Max', pronoun 'they');
+    $p2->verb('fly', 'flies');
+
+returns 'fly'
+
+=cut
+
+sub verb {
+	my ($self, $plur, $sing) = @_;
+	if (defined($sing)) {
+		return ($self->{pronoun} eq 'they' ? $plur : $sing);
+	} elsif (substr($plur, -1) eq 's' || substr($plur, -2) eq 'ch' || substr($plur, -2) eq 'sh') {
+		return ($self->{pronoun} eq 'they' ? $plur : $plur . 'es');
+	} elsif (substr($plur, -2) ne 'ay' && substr($plur, -1) eq 'y') {
```

Comment:
This handles `pay/pays`, but what about `key/keys`, `deploy/deploys`, and `buy/buys`? The exception to the `y->ies` rule should look for `[aeou]y`.

Reply:
Added code to handle these as well as well as test cases.